File: src/main/java/edu/psu/ist/plato/kaiming/arm64/Flag.java
package edu.psu.ist.plato.kaiming.arm64;

import edu.psu.ist.plato.kaiming.Machine.Arch;
import edu.psu.ist.plato.kaiming.MachFlag;

public enum Flag implements MachFlag {
    
    N(0),
    Z(1),
    C(2),
    V(3);

    public final int index;
    
    private Flag(int idx) {
        index = idx;
    }

    @Override
    public Arch arch() {
        return Arch.ARM64;
    }

    @Override
    public int index() {
        return index;
    }
    
}

File: src/main/scala/edu/psu/ist/plato/kaiming/arm64/Flag.scala
package edu.psu.ist.plato.kaiming.arm64

import edu.psu.ist.plato.kaiming.MachFlag
import edu.psu.ist.plato.kaiming.Machine

sealed abstract class Flag(val name: String, val index: Int) extends MachFlag {
  val arch = Machine.Arch.ARM64
}

object Flag {
  case object C extends Flag("C", 0)
  case object Z extends Flag("Z", 1)
  case object N extends Flag("N", 2)
  case object V extends Flag("V", 3)
}